packages/discord.js/src/structures/BaseGuildVoiceChannel.js

```diff
@@ -35,6 +53,40 @@ class BaseGuildVoiceChannel extends GuildChannel {
        */
       this.userLimit = data.user_limit;
     }
+
+    if ('video_quality_mode' in data) {
+      /**
+       * The camera video quality mode of the channel.
+       * @type {?VideoQualityMode}
+       */
+      this.videoQualityMode = data.video_quality_mode;
+    } else {
+      this.videoQualityMode ??= null;
+    }
+
+    if ('last_message_id' in data) {
+      /**
+       * The last message id sent in the channel, if one was sent
+       * @type {?Snowflake}
+       */
+      this.lastMessageId = data.last_message_id;
+    }
+
+    if ('messages' in data) {
+      for (const message of data.messages) this.messages._add(message);
+    }
+
+    if ('rate_limit_per_user' in data) {
+      /**
+       * The rate limit per user (slowmode) for this channel in seconds
+       * @type {number}
+       */
+      this.rateLimitPerUser = data.rate_limit_per_user;
+    }
+
+    if ('nsfw' in data) {
+      this.nsfw = Boolean(data.nsfw);
```

Comment:
```suggestion
      this.nsfw = data.nsfw;
```
it would seem like a Boolean conversion is not required here as if the property is present it should be a boolean already

Reply:
It's marked as [optional](https://discord.com/developers/docs/resources/channel#channel-object-channel-structure), so I guess it's better to be safe than sorry?

Reply:
we already check above if it is in the object. so at this point at least according to the documentation we should have a boolean here 100% of the time

Reply:
Yup, you are right, I overlooked that! This could be removed.